Template/Trigger/ElementVisibilityTrigger.web.js

```diff
@@ -310,10 +294,86 @@
             return false;
         }
 
+        function setMutationObserver(triggerEvent) {
+            return function () {
+                if (observeDomChanges && isMutationObserverSupported) {
+                    var targetNode = documentAlias.body;
+                    var config = {attributes: true, childList: true, subtree: true};
+                    observerMutation = new MutationObserver(function (mutationsList) {
+                        mutationObserverCallback(mutationsList, triggerEvent);
```

Comment:
@AltamashShaikh wonder if we could apply similar behaviour as in the scroll listener in https://github.com/matomo-org/tag-manager/blob/4.x-dev/javascripts/tagmanager.js#L609-L617

to only trigger this event like every 120ms so if this event is triggered eg multiple times per second or even like every ms then it won't slow down the site as much?

Reply:
Updated code to execute callback every 120ms as discussed, please check